tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/DisplayAnnotationTest.java

```diff
@@ -29,57 +29,65 @@
 import org.wso2.ballerinalang.compiler.tree.BLangClassDefinition;
 import org.wso2.ballerinalang.compiler.tree.BLangFunction;
 import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangService;
 import org.wso2.ballerinalang.compiler.tree.expressions.BLangExpression;
 import org.wso2.ballerinalang.compiler.tree.expressions.BLangInvocation;
 import org.wso2.ballerinalang.compiler.tree.expressions.BLangTypeConversionExpr;
 
 import java.util.List;
 
 /**
- * Class to test icon annotation.
+ * Class to test display annotation.
  *
  * @since 2.0
  */
-public class IconAnnotationTest {
+public class DisplayAnnotationTest {
 
     private CompileResult result;
     private CompileResult negative;
 
     @BeforeClass
     public void setup() {
-        negative = BCompileUtil.compile("test-src/annotations/icon_annot_negative.bal");
-        result = BCompileUtil.compile("test-src/annotations/icon_annot.bal");
+        negative = BCompileUtil.compile("test-src/annotations/display_annot_negative.bal");
+        result = BCompileUtil.compile("test-src/annotations/display_annot.bal");
     }
 
     @Test
-    public void testIconOnFunction() {
+    public void testDisplayAnnotOnFunction() {
         BLangFunction fooFunction = (BLangFunction) ((List) ((BLangPackage) result.getAST()).functions).get(0);
         BLangAnnotationAttachment annot = (BLangAnnotationAttachment) ((List) fooFunction.annAttachments).get(0);
-        Assert.assertEquals(getActualExpressionFromAnnotationAttachmentExpr(annot.expr).toString(), " {path: " +
-                "/fooIconPath.icon}");
+        Assert.assertEquals(getActualExpressionFromAnnotationAttachmentExpr(annot.expr).toString(),
+                " {iconPath: fooIconPath.icon,label: Foo function}");
+    }
+
+    @Test
+    public void testDisplayAnnotOnServiceDecl() {
+        BLangService service = (BLangService) result.getAST().getServices().get(0);
+        BLangAnnotationAttachment attachment = service.getAnnotationAttachments().get(0);
+        Assert.assertEquals(getActualExpressionFromAnnotationAttachmentExpr(attachment.expr).toString(),
+                " {iconPath: service.icon,label: service,misc: Other info}");
     }
 
     @Test (enabled = false)
```

Comment:
Can this be enabled?

Reply:
We can by removing ``.cloneReadOnly()` bit, but then when we fix the annotation cloning issue this will fail.

Reply:
That's OK, right? The rest will probably fail anyway so we can fix the set of tests together? WDYT?

Reply:
done